tmt/steps/execute/sti.py

```diff
@@ -0,0 +1,108 @@
+import os
+import click
+import tmt
+
+from tmt.utils import GeneralError
+
+# The default playbook used to run the tests
+DEFAULT_PLAYBOOK = 'tests/tests.yml'
+
+# Ansible inventory file required to run the tests
+ANSIBLE_INVENTORY = """
+[localhost]
+sut  ansible_host={hostname} ansible_user=root ansible_remote_tmp={remote_tmp}
+"""
+
+
+class ExecutorSTI(tmt.steps.execute.ExecutePlugin):
+    """ Run tests using ansible according to Standard Test Interface spec """
+    _doc = """
+    Execute STI tests
+
+    An inventory file is prepared according to the STI specification
+    and ansible-playbook is used to run the given playbook.
+    """
+
+    # Supported methods
+    _methods = [
+        tmt.steps.Method(
+            name='sti', doc=_doc, order=50),
+        ]
+
+    @classmethod
+    def options(cls, how=None):
+        """ Prepare command line options for given method """
+        options = []
+        if how == 'sti':
+            options.append(click.option(
+                '-p', '--playbook', metavar='PLAYBOOK',
+                help=f"""
+                    Ansible playbook to be run as a test.
+                    By default '{DEFAULT_PLAYBOOK}'
+                """))
+        return options + super().options(how)
+
+    def show(self):
+        """ Show discover details """
+        super().show(['playbook'])
+
+    def _playbook(self):
+        if 'playbook' in self.data and self.data['playbook']:
+            playbook = self.data['playbook']
+        else:
+            playbook = DEFAULT_PLAYBOOK
+
+        if not os.path.exists(playbook):
+            raise GeneralError(f"Playbook '{playbook}' not found.")
+
+        return playbook
+
+    def _inventory(self, guest):
+        """ Provides Ansible inventory compared to STI """
+        hostname = guest.guest or guest.container
+
+        inventory_file = os.path.join(
+            self.step.workdir, f'inventory-{hostname}')
```

Comment:
You can use `self.step.write(f'inventory-{hostname}', content)` to create a file in the step workdir.